internal/provider/models/processors/aggregate_v2.go

```diff
@@ -85,74 +78,47 @@ var AggregateV2ProcessorResourceSchema = schema.Schema{
 	}),
 }
 
-func methodConfigFromModel(plan *AggregateV2ProcessorModel, userConfig map[string]any, dd *diag.Diagnostics) {
-	isIntervalSet := !plan.Interval.IsNull() && !plan.Interval.IsUnknown()
-	isDurationSet := !plan.Duration.IsNull() && !plan.Duration.IsUnknown()
+func windowConfigFromModel(plan *AggregateV2ProcessorModel, userConfig map[string]any, dd *diag.Diagnostics) {
 	isMinimumSet := !plan.Minimum.IsNull() && !plan.Minimum.IsUnknown()
-	method := plan.Method.ValueString()
-
-	userConfig["method"] = method
-	if method == "tumbling" {
-		if isDurationSet {
-			dd.AddError(
-				"Error in plan",
-				"The field 'window_duration' can only be set if method == 'sliding'",
-			)
-		}
-
-		if isMinimumSet {
-			dd.AddError(
-				"Error in plan",
-				"The field 'window_min' can only be set if method == 'sliding'",
-			)
-		}
-
-		if isIntervalSet {
-			userConfig["interval"] = plan.Interval.ValueInt64()
-		}
-	} else if method == "sliding" {
-		if isIntervalSet {
-			dd.AddError(
-				"Error in plan",
-				"The field 'interval' can only be set if method == 'tumbling'",
-			)
-		}
 
-		if isDurationSet {
-			userConfig["window_duration"] = plan.Duration.ValueInt64()
-		}
-		if isMinimumSet {
-			userConfig["window_min"] = plan.Minimum.ValueInt64()
-		}
-	} else {
+	windowConfig := make(map[string]any)
+	windowConfig["type"] = plan.WindowType.ValueString()
+	windowConfig["interval"] = plan.Interval.ValueInt64()
+	if windowConfig["type"] == "tumbling" && isMinimumSet {
 		dd.AddError(
 			"Error in plan",
-			"The method '%s' is not handled correctly in the provider. Please open a GitHub issue to report this.",
+			"The field 'window_min' is only allowed when using a sliding window type.",
 		)
+	} else {
+		if isMinimumSet {
+			windowConfig["window_min"] = plan.Minimum.ValueInt64()
+		}
 	}
+	userConfig["window"] = windowConfig
 }
 
-func strategyConfigFromModel(plan *AggregateV2ProcessorModel, userConfig map[string]any, dd *diag.Diagnostics) {
-	strategySet := !(plan.Strategy.IsNull() || plan.Strategy.IsUnknown())
-	scripSet := !(plan.Script.IsNull() || plan.Script.IsNull())
-	if !strategySet && !scripSet {
+func evaluateConfigFromModel(plan *AggregateV2ProcessorModel, userConfig map[string]any, dd *diag.Diagnostics) {
+	isOperationSet := !(plan.Operation.IsNull() || plan.Operation.IsUnknown())
+	isScripSet := !(plan.Script.IsNull() || plan.Script.IsNull())
```

Comment:
should one of these be `plan.Script.IsUnknown()` ? (I realize you just moved/renamed this declaration)

Reply:
Good catch. I'll fix it.